### PR TITLE
Azure Site Recovery (ASR) - Customer managed keys based encryption support.

### PR DESCRIPTION
**Pull request by ayfathim**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ayfathim** [comment]
ModelValidationFailures and Semantic Validation failures are nothing to do with the change we have introduced as part of this PR / Commit.

@yungezz : please review

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 1]
```diff
@@ -7799,6 +7799,10 @@
           "description": "The target disk type after failover. Its an optional value and will be same as source disk type if not user provided.",
           "type": "string"
         },
+        "recoveryDiskEncryptionSetId": {
```
Comment (yungezz) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: add  new property in stable version will cause breaking change, suggest to add in new version.

Comment (ayfathim) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: This is an optional parameter. And not a breaking change.

Comment (ayfathim) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: These newly added properties are option (not required properties). As per PR #6380(https://github.com/Azure/azure-rest-api-specs/pull/6380/files) , it is required to bump up the API version only on adding required properties.

Comment (ayfathim) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: @yungezz : Please add the required person , who will be doing the ARM review.

**Review by yungezz — APPROVED** [review]

**Comment by ayfathim** [comment]
@raych1: I am still getting semantic validation failures, even after adding the suppression. Can you please us here.

File: specification/recoveryservicessiterecovery/resource-manager/readme.md [hunk 2]
```diff
@@ -33,6 +33,11 @@ directive:
       - $.paths
     suppress:
       - UniqueResourcePaths
+
+directive:
+  - from: notRequiredDiscriminatorWithSuppression.json
```
Comment (raych1) [inline specification/recoveryservicessiterecovery/resource-manager/readme.md]: The value of `from` should be the file name you're going to suppress. I assume it is service.json in this case.

**Comment by raych1** [comment]
@ayfathim , suppression config is not right. Please see the comment.

File: specification/recoveryservicessiterecovery/resource-manager/readme.md [hunk 3]
```diff
@@ -33,6 +33,11 @@ directive:
       - $.paths
     suppress:
       - UniqueResourcePaths
+
+  - from: service.json
```
Comment (raych1) [inline specification/recoveryservicessiterecovery/resource-manager/readme.md]: This should defined in individual suppression section like below.
```
## Suppression

``` yaml
directive:
  - from: service.json
    suppress: OAV131 
    reason: Testing purpose.
```

Comment (ayfathim) [reply specification/recoveryservicessiterecovery/resource-manager/readme.md]: added separate section

**Comment by ayfathim** [comment]
@raych1: added the suppression for semantic validation failures. Please  approve and merge the PR.

**Comment by raych1** [comment]
@yungezz , as talked with @ayfathim , they have task created in backlog to fix the examples. It's ok to merge this PR since the model validation failures are not caused by this PR changes, thanks.

**Pull request merged**